hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/ContainerStateMachine.java

```diff
@@ -1006,7 +1006,7 @@ private void removeStateMachineDataIfNeeded(long index) {
         RaftServer.Division division = ratisServer.getServer().getDivision(getGroupId());
         if (division.getInfo().isLeader()) {
           long minIndex = Arrays.stream(division.getInfo()
-              .getFollowerNextIndices()).min().getAsLong();
+              .getFollowerNextIndices()).min().orElse(0);
           LOG.debug("Removing data corresponding to log index {} min index {} "
                   + "from cache", index, minIndex);
           removeCacheDataUpTo(Math.min(minIndex, index));
```

Comment:
Thanks @jojochuang for the patch.

Should it `removeCacheDataUpTo` only if `minIndex` is found?

```java
Arrays.stream(division.getInfo().getFollowerNextIndices())
    .min()
    .ifPresent(minIndex -> removeCacheDataUpTo(Math.min(minIndex, index)));
```

Also, would be nice to rethrow unchecked exceptions directly:

```java
try {
  ...
} catch (RuntimeException e) {
  throw e;
} catch (Exception e) {
  throw new RuntimeException(e);
}
```

Reply:
I think it's essentially the same, but your suggest is more readable. Will update accordingly.